docs/src/examples/components/Popup/Variations/PopupExamplePosition.shorthand.tsx

```diff
@@ -1,145 +1,67 @@
 import React from 'react'
 import { Button, Grid, Popup } from '@stardust-ui/react'
 
+class PopupArrowExample extends React.Component<any, any> {
+  state = { popupOpen: false }
+
+  togglePopupState = () => {
```

Comment:
you removed the examples that show that the popup state (open/closed reflected by `open` prop) can also be handled by the popup internally without adding an event handler for the trigger; we should keep that example since `Popup` is now an autocontrolled component

Reply:
sorry - you mean, to add two examples with different values provided to the `open` prop?

Reply:
I mean examples like this one below where we do not have to add custom logic or handlers for handling basic popup functionality; this used to work in SUIR and I believe we should still have it

```
import React from 'react'
import { Button, Popup } from '@stardust-ui/react'

const PopupExample = () => (
  <Popup trigger={<Button icon="expand" />} content="Add users to your feed." />
)

export default PopupExample
```

Reply:
Here is the problem - there is a need to make Popup to be a controlled component. Stardust suggests to achieve this by extending `AutoControlledComponent` type (for now), and this type enforces controlled mode for the component: https://reactjs.org/docs/forms.html#controlled-components

Any attempt to duplicate `open` prop in state of the `Popup` (and, note, that this is essentially the only way to handle `open` state changes by `Popup` itself) will raise an error, and this policy is enforced by `AutoControlledComponent` by intent.

Thus, _for now_ only controlled mode is supported, and this mode will provide all the necessary functionality for the client. Still, we would be able to extend the approach taken and introduce uncontrolled mode to `Popup` in future - specifically, once `AutoControlledComponent` approach will be reconsidered.

Reply:
ah, so the uncontrolled mode was taken down because of the new `AutoControlledComponent` approach was not decided; I thought it was an oversight, got it; thx!